internal/hcl/evaluator.go

```diff
@@ -115,7 +115,9 @@ func NewEvaluator(
 	isGraph bool,
 ) *Evaluator {
 	ctx := NewContext(&hcl.EvalContext{
-		Functions: ExpFunctions(module.RootPath, logger),
+		// Functions require the module path to be passed in rather than the RootPath
+		// as the latter will not resolve correct relative paths for file functions.
+		Functions: ExpFunctions(module.ModulePath, logger),
```

Comment:
This resolves a hidden bug that meant relative paths used in file functions in modules would always fail as the `basePath` was always the top level `RootPath`. E.g. a project with the following structure:

```
.
└── terraform
    ├── main.tf
    └── modules
        └── foo
            ├── file.json
            └── main.tf
```

The child `Evaluator` for module `foo` would use the `basePath` `terraform` in `ExpFunctions`. Meaning that if a the hcl tried to open a file.json e.g.`file("file.json")` we would attempt to open `terraform/file.json` which did not exist. 

Passing the `ModulePath` means that the base path resolves to `terraform/module/foo`, solving the relative filepath issue.